pkg/batch/executor_test.go

```diff
@@ -11,6 +11,50 @@ import (
 	"github.com/treeverse/lakefs/pkg/logging"
 )
 
+const accessedOnce = 1
+const accessedTwice = 2
+
+type reader struct {
+	Ch         chan bool
+	IsExecuted bool
+	Fn         ExecFn
```

Comment:
Think that adding `Fn` callback as part of the reader which is a callback can make the code a bit complex.
Based on the use of this function I suggest just adding another channel, one that the reader will close when it is executed.